### ISSUE TITLE
Implement The End

### ISSUE THREAD

**Issue by Bond-009** [cuberite/cuberite#4056]
Client version:   /
Server OS:        /
Commit id:        76a0971279de47c5a1d8510848e88e57cc6eed72

### Expected behavior
![image](https://user-images.githubusercontent.com/21289123/31186002-b6e0fb3e-a92d-11e7-900b-ca1567415fd0.png)

### Actual behavior
![screenshot from 2017-10-04 17-56-26](https://user-images.githubusercontent.com/21289123/31186011-bec27602-a92d-11e7-8767-8555a6c10da9.png)

### Steps to reproduce the behavior
Go to The End

**Comment by 12xx12** [comment]
**Things to do:**
```
-[ ] Terrain generation
    - [ ] initial island
    - [ ] the nothing
    - [ ] the further islands

- [ ] Initial island
    - [ ] obsidian towers for dragon fight
    - [ ] spawn plattfor for player
    - [ ] spawn the dragon

- [ ] Other islands
    - [ ] the villages
    - [ ] chorus flowers (tree like generation)

- [ ] enderman spawning
```

**Comment by 12xx12** [comment]
Updates in #4968

**Comment by PureTryOut** [comment]
The spawn point is at the moment also quite broken. I just created a custom end portal and went to the end, but spawned on 0, 0, 0 and immediately started falling to the void.

Instead it should always spawn you near the middle island and with a small (I believe 4x4?) obsidian platform beneath you.

**Comment by 12xx12** [comment]
You are right. This needs to be done in a seperate PR because the wiki states that the plattform is generated every time the player enters the end dimension and it's always at the same coordinates. So the end dimension always has the same spawn coordinates on initial generation.

**Comment by PureTryOut** [comment]
Yeah I noticed that it regenerates the platform every time as I tried to create a structure on it once and it partially disspeared when I respawned.
